contracts/Router.sol

```diff
@@ -50,6 +50,8 @@ contract Router {
     address private charity;
     uint256 private charityCut;
     uint256 private rewardsCut;
+    unit256 private oldReserveBalance;
+    bool private firstRebase;
```

Comment:
instead of having to set `firstRebase` to true in constructor, we can instead change it to `hasTransferredOldReserveBalance` and it will remain false until the old reserve balance is transferred, at which point we set to `true` for future rebases

Reply:
Also `olderReserveBalance` should be `uint256` not `unit256`